File: 11-ContainerWithMostWater/11-ContainerWithMostWater.cpp
// Last updated: 7/16/2026, 4:39:31 PM
class Solution {
public:
    int maxArea(vector<int>& height) {
        int i  = 0;
        int j = height.size()-1;
        
        int area = 0;
        while(i<j){
        int w = j - i;
        int h= min(height[i], height[j]);
        area = max(area, w * h);

        if(height[i]>height[j]){
            j--;
        }else i++;
        }
        return area;
    }
    
};

File: 11-ContainerWithMostWater/11-ContainerWithMostWater.py
# Last updated: 7/16/2026, 4:39:37 PM
class Solution(object):
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """
        ans=0
        i=0
        j=len(height)-1
        while (i < j):
            if height[i]<=height[j]:
                area = height[i]*(j - i)
                i += 1
            else:
                area = height[j]*(j - i)
                j -= 1
            if area > ans:
                 ans = area
        return ans